packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -640,4 +644,95 @@ export class CallContext {
     this._atomicId++;
     return id;
   }
+
+  /**
+   * Tees any errors encountered in an async function to the state.
+   *
+   * If the function succeeds, clears associated errors from the state.
+   *
+   * @param f Async function to execute.
+   * @param target The error target to tee error to.
+   * @param clearTargets The error targets to clear errors for if the function succeeds. By default, clears errors for `target`.
+   * @returns Result of calling `f`. Also re-raises any exceptions thrown from `f`.
+   * @throws CallError. Exceptions thrown from `f` are tagged with the failed `target.
+   */
+  public withAsyncErrorTeedToState<Args extends unknown[], R>(
+    f: (...args: Args) => Promise<R>,
```

Comment:
nit: personally don't like the single letter name for a parameter as it's not explicit at all
asyncAction or something around it might be better

Reply:
Done here, but: I actually really like single letter names for local variables, and sometimes function parameters -- it means that your functions are so short, and named such that you don't need any context to understand what that variable / parameter means.

In this example `withAsyncErrorTeedToState`, takes exactly one argument, the function to wrap. The type of the function to wrap is important (tells you that it's a function), but the name doesn't help all that much.
Also, the implementation is short enough that `f` suffices (in my opinion).

Also, named it `action`, not `asyncAction`, the `async` part is in the type.